magma/frontend/coreir_.py

```diff
@@ -1,21 +1,28 @@
 from magma import cache_definition
-from magma.backend.coreir_ import CoreIRBackend, CoreIRContextSingleton
+from magma.backend.coreir.coreir_backend import CoreIRBackend
+from magma.backend.coreir.coreir_runtime import (coreir_context,
+                                                 reset_coreir_context)
 from magma.circuit import DefineCircuitKind, Circuit
 from magma import cache_definition, Clock, Array, BitIn, BitOut, Product
 from coreir.generator import Generator
 
+
 @cache_definition
 def GetCoreIRBackend():
     return CoreIRBackend()
 
+
 def GetMagmaContext():
-    return CoreIRContextSingleton().get_instance()
+    return coreir_context()
+
 
 def ResetCoreIR():
-    CoreIRContextSingleton().reset_instance()
+    reset_coreir_context()
     backend = GetCoreIRBackend()
+    # NOTE(rsetaluri): Is this next line necessary? Don't think it is.
     backend.reset()
```

Comment:
This calls the backend._init function which resetts all the dictionaries/sets which I think is necessary.  For example, the constant cache holds references to modules for constants.  If we reset the context, we need to clear the backend context.

Reply:
Updated comment.